tensorboard/webapp/widgets/card_fob/card_fob_controller_test.ts

```diff
@@ -40,22 +48,27 @@ class TestableComponent {
 
   @Input() axisDirection!: AxisDirection;
   @Input() timeSelection!: TimeSelection;
-  @Input() cardFobAdapter!: CardFobAdapter;
+  @Input() cardFobHelper!: CardFobGetStepFromPositionHelper;
   @Input() showExtendedLine?: Boolean;
 
   @Input() onTimeSelectionChanged!: (newTimeSelection: TimeSelection) => void;
   @Input() onTimeSelectionToggled!: () => void;
+  @Input() getAxisPositionFromStartStep!: () => number;
```

Comment:
Can you change the types of these four new properties from function to plain-old number? And refactor the tests to reflect this?

Reply:
I assume you could do the following on L123 to 128:
```

    fixture.componentInstance.axisPositionFromStartStep = input.timeSelection.start.step;
    fixture.componentInstance.axisPositionFromEndStep = input.timeSelection.end ? input.timeSelection.end.step : null;
    fixture.componentInstance.highestStep = 4;
    fixture.componentInstance.lowestStep = 0;
```

Reply:
With assigning to axisPositionFromStartStep directly, the tests are failing. (for example the first test below "vertical dragging" at line 166 and line 182 fails)
After debugging I found the reason is that although the initial value is taken from input.timeSelection, the timeSelection variable got updated from mousemove, but it never updates "`startStepAxisPosition`" and "`getCssTranslatePxForStartFob`" still returns the same initial number.

I think highestStep and lowestStep can be assigned from input but `axisPositionFromStartStep` and `axisPositionFromEndStep` should be kept as function and check the spy. 

Let me know if you have any other suggestions.

Reply:
Oh I see. Yes, that makes sense! Thanks for explaining.